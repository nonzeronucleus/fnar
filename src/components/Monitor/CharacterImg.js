import React from 'react';
import styled from 'styled-components';
import characterImages from '../../img/characters'

const CharacterImg = styled.img`
    position: absolute;
    top:0px;
    left: ${props => props.pos*180}px;
    z-index: 2;
`;

export default ({character, pos}) => <CharacterImg src={characterImages[character]} role="presentation" alt="" id="room" />
