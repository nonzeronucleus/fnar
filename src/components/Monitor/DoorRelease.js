import React from 'react';
import styled from 'styled-components';
import switchImage from '../../img/switch/switch-off.png';

const SwitchImg = styled.img`
    position: absolute;
    top:400px;
    left:400px;
    z-index:4;
`;

export default () => <SwitchImg src={switchImage} />
