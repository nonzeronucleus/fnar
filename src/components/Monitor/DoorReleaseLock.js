import React from 'react';
import styled from 'styled-components';
import lockClosed from '../../img/rooms/lock-closed.png';
import lockOpen from '../../img/rooms/lock-open.png'

const LockImage = styled.img`
    position: absolute;
    top:200px;
    left:400px;
    z-index:4;
`;

export default ({active}) => (
    <LockImage src={active ? lockClosed : lockOpen} />
)
