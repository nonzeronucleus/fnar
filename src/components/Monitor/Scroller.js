import React from 'react';
import styled, {keyframes} from 'styled-components';


const entry = keyframes`
    from {
        transform: rotateX(-90deg);
        transform-origin: bottom;
    }
    to {
        transform: rotateX(0);
        transform-origin: bottom;
    }
`;

const swing = keyframes`
    from {left : -400px;}
    to {left : 0px;}
`;


const Frame = styled.div`
    width:1200px;
    height:768px;
    overflow: hidden;
    position:absolute;
    top:0px;
    left:0px;

    transition: text-indent 1s;
`;



const Inner = styled.div`
    width:1920px;
    height:768px;
    overflow: hidden;
    background-color:blue;
    animation: ${entry} 0.2s, ${swing} 8s ease-in-out infinite;
    animation-direction: alternate;
    position:absolute;
    transform-style: preserve-3d;
`;


export default ({scroll, children}) => {
    return(
        <Frame>
            <Inner>
                {children}
            </Inner>
        </Frame>
    );
}

