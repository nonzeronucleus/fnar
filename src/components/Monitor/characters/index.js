import characters from '../../../consts/characters'

import ginger from './Ginger.jpg';
import merwing from './Merwing.jpg';
import minty from './Minty.jpg';
import porkie from './Porkie.jpg';
import rainba from './rainba.jpg';

export default {
    [characters.GINGER]: ginger,
    [characters.MERWING]: merwing,
    [characters.MINTY]: minty,
    [characters.PORKIE]: porkie,
    [characters.RAINBA]: rainba,
}
