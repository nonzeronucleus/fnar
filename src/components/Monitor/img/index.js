// import Hat1 from '../../assets/img/accesories/hats/hat1.png';
// import Hat2 from '../../assets/img/accesories/hats/hat2.png';
// import Hat3 from '../../assets/img/accesories/hats/hat3.png';
// import Hat4 from '../../assets/img/accesories/hats/hat4.png';
// import Hat5 from '../../assets/img/accesories/hats/hat5.png';
// import Hat6 from '../../assets/img/accesories/hats/hat6.png';

// export const hatsArr = [
//     Hat1,
//     Hat2,
//     Hat3,
//     Hat4,
//     Hat5,
//     Hat6
// ];