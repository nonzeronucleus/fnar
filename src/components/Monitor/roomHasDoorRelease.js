import releaseButtons from '../../consts/releaseButtons';

export default (room) => !!releaseButtons[room];