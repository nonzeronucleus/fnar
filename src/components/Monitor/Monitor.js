import React, {useCallback}  from 'react';
import {useMappedState} from 'redux-react-hook';
import styled from 'styled-components';
import {getSelectedRoom, getCharactersInSelectedRoom} from '../../redux/selectors';
import roomImages from '../../img/rooms';
import RadarMap from '../RadarMap';
import cameraImage from '../../img/rooms/camera-effect.gif'
import roomHasDoorRelease from './roomHasDoorRelease';
import Scroller from './Scroller'
import DoorRelase from './DoorRelease';
import CharacterImg from './CharacterImg';

const RoomImg = styled.img`
  position: absolute;
  top:0px;
  z-index: 2;
`;

const CameraImage = styled.img`
    position:absolute;
    top:0px;
    left:0px;
    width:1600px;
    height:768px;
    z-index:3;
    opacity:0.6;
`;

export default () => {
    const mapState = useCallback(
        state => ({
          room: getSelectedRoom(state),
          charactersInRoom: getCharactersInSelectedRoom(state),
        }), []
    );

    const {room, charactersInRoom} = useMappedState(mapState);

    return (
        <>
            <RadarMap />
            <CameraImage src={cameraImage} />

            <Scroller>
                <RoomImg src={roomImages[room]} role="presentation" alt="" id="room"/>
                {roomHasDoorRelease(room) && <DoorRelase />}
                { charactersInRoom.map( (character, i)  => {
                    return <CharacterImg {...{character}} key={i} role="presentation" alt="" id="room" pos={i}/>
                })}
            </Scroller>
        </>
    )
}
