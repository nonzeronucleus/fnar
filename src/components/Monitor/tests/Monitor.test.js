import React from 'react';
import Monitor from '../Monitor';
import renderer from 'react-test-renderer';
import {useMappedState} from 'redux-react-hook';

jest.mock('redux-react-hook', () => {
  return {
    useMappedState: jest.fn()
  }
});

jest.mock('../../RadarMap', () => ()=> (<div>A</div>))

it('renders correctly with a character in the room', () => {

  useMappedState.mockImplementation(() => ({
    selectedRoom:"dining room",
    charactersInRoom:[
      "rainba"
    ]
  }))

  const tree = renderer
    .create(
        <Monitor/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly with no characters in the room', () => {

  useMappedState.mockImplementation(() => ({
    selectedRoom:"dining room",
    charactersInRoom:[
    ]
  }))

  const tree = renderer
    .create(
        <Monitor/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
