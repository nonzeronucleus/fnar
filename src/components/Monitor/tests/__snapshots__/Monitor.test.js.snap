// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a character in the room 1`] = `
Array [
  <div>
    A
  </div>,
  <img
    className="sc-EHOje cgBQfU"
    src="camera-effect.gif"
  />,
  <div
    className="sc-bdVaJa fJRvkx"
  >
    <div
      className="sc-bwzfXH cHplRP"
    >
      <img
        alt=""
        className="sc-ifAKCX dhCAyy"
        id="room"
        role="presentation"
      />
      <img
        alt=""
        className="sc-bxivhb iVyuZR"
        id="room"
        role="presentation"
      />
    </div>
  </div>,
]
`;

exports[`renders correctly with no characters in the room 1`] = `
Array [
  <div>
    A
  </div>,
  <img
    className="sc-EHOje cgBQfU"
    src="camera-effect.gif"
  />,
  <div
    className="sc-bdVaJa fJRvkx"
  >
    <div
      className="sc-bwzfXH cHplRP"
    >
      <img
        alt=""
        className="sc-ifAKCX dhCAyy"
        id="room"
        role="presentation"
      />
    </div>
  </div>,
]
`;
